src/lib/utils/tensors.ts

```diff
@@ -97,3 +97,23 @@ export const ensure2DMatrix = (X: Type2DMatrix<number> | Type1DMatrix<number>):
   const matrix1D = validateMatrix1D(X);
   return _.map(matrix1D, (o) => [o]);
 };
+
+export const covariance = (
```

Comment:
We should add unit tests and a docstring with @ignore so it won't show up on the documentation